### PR TITLE
More informative error messages

### LINKED ISSUE #111: Clarify error message for "forbidden_modifications"
At the moment if you have DESCRIPTION with uncommitted changes and try to run `fledge::bump_version()`, the error message is not explicit.

I'd expect some clear instructions à la https://github.com/r-lib/usethis/blob/3385e148c06b13a34406a455d7a7ab0785993201/R/utils-git.R#L156

### LINKED ISSUE #134: Use a variant for the snapshot test that depends on testthat?
https://www.tidyverse.org/blog/2021/10/testthat-3-1/

"expect_snapshot() and friends get an experimental new variant argument which causes the snapshot to be saved in _snaps/{variant}/{test}.md instead of _snaps/{test}.md"

(not exactly needed in fledge case where we can hope the test will get stable once rlang current dev version is released, but interesting as you @krlmlr used a variant of the approach :grin: )

### PR DESCRIPTION
**Pull request by maelle**
Fix #111
Fix #134 (unplanned :sweat_smile: )

### CODE REVIEW

**Comment by maelle** [comment]
Other candidates for improvements, but not in scope for this PR, are other `stopifnot()`.

File: tests/testthat/_snaps/bump_version.md [hunk 1]
```diff
@@ -26,31 +26,12 @@
     Output
       NULL
 
-# bump_version() works (CRAN rlang)
```
Comment (maelle) [inline tests/testthat/_snaps/bump_version.md:29]: rlang was released

Comment (krlmlr) [reply tests/testthat/_snaps/bump_version.md:29]: Do we have a versioned dependency on rlang >= 0.4.12 in `DESCRIPTION`?

File: R/commit-version.R [hunk 2]
```diff
@@ -44,7 +44,19 @@ get_commit_message <- function(version) {
...
+  unexpected <- forbidden_modifications[forbidden_modifications %in% status$file]
+
+  if (length(unexpected) == 0){
+    return()
+  }
+
+  rlang::abort(
+    message = sprintf(
+      "Unindexed change(s) in %s, commit them before running any fledge function again.",
+      toString(unexpected)
```
Comment (krlmlr) [inline R/commit-version.R:56]: Do we want markup for the file names?

**Review by krlmlr — COMMENTED** [review]
Thanks, looks good!

File: tests/testthat/_snaps/bump_version.md (hunk 1 above)
Comment (maelle) [reply tests/testthat/_snaps/bump_version.md:29]: Now we do, good catch :grin:

File: tests/testthat/test-bump_version.R [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
-
   tempdir <- withr::local_tempdir(pattern = "fledge")
   repo <- create_demo_project(open = FALSE, dir = tempdir, news = TRUE)
   usethis::with_project(
     path = repo, {
       use_r("bla")
       gert::git_add("R/bla.R")
       gert::git_commit("* Add cool bla.")
-      expect_snapshot(bump_version())
+      expect_snapshot(bump_version(), variant = as.character(packageVersion("rlang")))
```
Comment (krlmlr) [inline tests/testthat/test-bump_version.R:9]: Nice idea, it seems that using the version directly is not sustainable. Maybe something like:

```suggestion
      expect_snapshot(bump_version(), variant = if (packageVersion("rlang") > "0.99") "rlang-1" else "rlang-legacy")
```

Maybe refactor out the rlang variant to a function?

**Review by krlmlr — COMMENTED** [review]
Thanks! Let's do another iteration here, this seems minor enough to not conflict the other PRs.

File: tests/testthat/test-bump_version.R (hunk 3 above)
Comment (maelle) [reply tests/testthat/test-bump_version.R]: Done!

**Comment by maelle** [comment]
@krlmlr does this look ok? I'll use the test variants for https://github.com/cynkra/fledge/pull/157#discussion_r748431930

**Review by krlmlr — APPROVED** [review]
Looks good, thanks. Eventually we might want to implement a wrapper. How to ensure we don't forget to clean up unused variants, e.g. when rlang 1.0.0 is released?
